src/sentry/static/sentry/app/views/usageStats/usageChart/utils.tsx

```diff
@@ -1,26 +1,40 @@
 import moment from 'moment';
 
-import {DataCategory} from 'app/types';
+import {DataCategory, IntervalPeriod} from 'app/types';
 
 import {formatUsageWithUnits} from '../utils';
 
-export function getDateFromMoment(m: moment.Moment) {
-  return m.format('MMM D');
+export function getDateFromMoment(m: moment.Moment, interval: IntervalPeriod = '1d') {
+  const unit = interval.replace(/[0-9]/g, '');
+
+  switch (unit) {
+    case 'd':
+      return m.startOf('h').format('MMM D');
+    default:
+      return m.startOf('h').format('MMM D HH:mm');
+  }
 }
 
 export function getDateFromUnixTimestamp(timestamp: number) {
   const date = moment.unix(timestamp);
   return getDateFromMoment(date);
 }
 
-export function getDateRange(dateStart: string, dateEnd: string): string[] {
+export function getXAxisDates(
+  dateStart: string,
+  dateEnd: string,
+  interval: IntervalPeriod = '1d'
+): string[] {
   const range: string[] = [];
-  const start = moment(dateStart);
-  const end = moment(dateEnd);
+  const start = moment(dateStart).startOf('h');
+  const end = moment(dateEnd).startOf('h');
+
+  const amount = Number(interval.replace(/[a-zA-Z]/g, ''));
+  const unit = interval.replace(/[0-9]/g, '');
```

Comment:
There's a parsing func in 
![image](https://user-images.githubusercontent.com/79684/113604097-a609fd80-95f9-11eb-850f-2c28ff838847.png)